tools/summaries/build_ranking_based_summaries.py

```diff
@@ -0,0 +1,216 @@
+                                  stat_var_json: str, output_file: str):
+  """Get a summary for all child places of a parent place, based on rankings.
+  
+  Builds a summary using templated sentences. A place's observations for a
+  variable for that place is highlighted if they rank highly for that place.
+
+  Args:
+    place_type: the place type of the child places to write summaries about
+    parent_place_dcid: dcid of the parent place
+    stat_var_json: path to a stat var config file, listing stat vars to use
+    output_file: path to write output to
+  """
+  child_places = dc.get_child_places(place_type, parent_place_dcid)
+  name_of = dc.get_names(child_places + [parent_place_dcid])
+  parent_place_name = name_of[parent_place_dcid]
+  if parent_place_dcid == "country/USA":
+    # USA needs "the" in front in sentences.
+    parent_place_name = "the United States of America"
+  #population_of = get_population(child_places)
+  sentences = initialize_summaries(child_places, name_of, place_type,
+                                   parent_place_name)
+
+  # Process each variable
+  with open(stat_var_json) as sv_f:
+    sv_data = json.load(sv_f)
+
+    for category, sv_list in sv_data.items():
+
+      if category in _CATEGORY_SKIP_LIST:
+        continue
+
+      for sv in sv_list:
+        # Get ranking for variable
+        rank_list = dc.get_ranking_by_var(stat_var_dcid=sv["sv"],
+                                          place_type=place_type,
+                                          parent_place_dcid=parent_place_dcid)
+
+        # Add summaries for top places
+        for i in range(len(rank_list)):
+          rank_item = rank_list[i]
+          place_dcid = rank_item['placeDcid']
+          sv_value = format_stat_var_value(value=rank_item['value'],
+                                           stat_var_data=sv)
+          sentence = None
+
+          if i < _DEFAULT_RANKING_THRESHOLD:
+            sentence = _TEMPLATE_RANKING_SENTENCE.format(
+                place_name=name_of[place_dcid],
+                rank=get_rank_string(rank_item['rank']),
+                parent_place_name=parent_place_name,
+                stat_var_name=sv['name'],
+                value=sv_value,
+            )
+
+          elif i <= len(rank_list) * _DEFAULT_PERCENTAGE_THRESHOLD:
+            sentence = _TEMPLATE_VALUE_SENTENCE.format(
+                stat_var_name=sv['name'],
+                place_name=name_of[place_dcid],
+                value=sv_value,
+                date_str="")
+
+          if sentence:
+            sentences[place_dcid].append(sentence)
+
+  # Combine sentences into a paragraph:
+  summaries = {}
+  for place, sentence_list in sentences.items():
+    summaries[place] = {"summary": " ".join(sentence_list)}
+
+  # Write summaries to file
+  with open(output_file, "w") as out_file:
+    json.dump(summaries, out_file, indent=4)
+
+
+def main():
+  build_ranking_based_summaries(place_type="State",
```

Comment:
For a future PR: take these in as command line arguments (look into using [click](https://click.palletsprojects.com/en/8.1.x/))

Reply:
Thanks for the pointer! Will add in next PR.